src/plugins/data_explorer/public/ui_metric/constants.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const NEW_DISCOVER_LOAD_EVENT = 'new_discover_load_count';
```

Comment:
super duper nit: don't change this PR specifically for this but the import line for src/plugins/data_explorer/public/components/app_container.tsx could be a smaller and not require future updates if we utilize an object. and easier to access for developers. and probably should keep the casing and the same.

suggestion like:

```typescript
const APP_NAME = `discover`;
const LOAD_EVENT = `load_count`;
const OPT_IN_EVENT = `opt_in_count`;
const OPT_OUT_EVENT = `opt_out_count`;
export const METRIC = {
  DEFAULT: {
     LOAD_EVENT: `${APP_NAME}_${LOAD_EVENT}`,
     OPT_IN_EVENT: `${APP_NAME}_${OPT_IN_EVENT}`,
     OPT_OUT_EVENT: `${APP_NAME}_${OPT_OUT_EVENT}`
  },
  LEGACY: {
     LOAD_EVENT: `${APP_NAME}_legacy_${LOAD_EVENT}`,
     OPT_IN_EVENT: `${APP_NAME}_legacy_${OPT_IN_EVENT}`,
     OPT_OUT_EVENT: `${APP_NAME}_legacy_${OPT_OUT_EVENT}`
  }
}
```

then we can just import `METRIC` and access the constant with the dot notation.

some things to mention is that proposing using `legacy` instead of classifying vs using the `new` because with new or utilizing the `2.0` is that we version this ui metric to be 2.0 and then if we make the new discover the default then it becomes confusing for those who don't have context that the the default is the `new` ui metric and the previous version was the one without any classifying word on the ui metric. however, im not super tied to calling legacy. but i think we should go with legacy or new and avoid the `2.0` or `2`.

also opted to ensure all the metrics are the same structure like

```typescript
appname_classifier_event_metrictype
```

classifier being like `new` or `legacy` or nothing. this way if other plugins utilize the same standard then we have a nice way to aggregate on the results in a similar structure for visualizations and decision making. vs writing a query that filters out only the new and making the regex be something so it filters for `new_discover` or `discover_new`.

the final is using of interpolation this way if another developer wants to track another metric they can add by seeing the standard of using interpolation and do it as well. the reason why because i've seen people hardcode type constants over and over again and then there was a typo at which point we used that in key for local storage. but since the users used it already we are kind of stuck. so if we reduce the amount of opportunities for someone to make a typo then we could avoid an issue like queries failing to show up if the query was `appName: discover` but the developer added a new metric `discoever_refresh_count` where they mixed the `e` and the `v` in discover.